x5424/logger.go

```diff
@@ -134,20 +137,43 @@ func (l *Logger) Log(v ...interface{}) error {
 
 func (l *Logger) log(m map[string]uint, v ...interface{}) error {
 	var format []string
+	var colorOut *color.Color
+
 	if facility, ok := m[Facility]; ok {
 		format = append(format, l5424.FacilityLvl(facility).String())
+		if l.color {
+			switch l5424.FacilityLvl(facility).String() {
+			case "FacilityAlert":
+				colorOut = color.New(color.FgRed, color.Bold)
+			}
+		}
 	}
 	if severity, ok := m[Severity]; ok {
 		format = append(format, l5424.SeverityLvl(severity).String())
+		if l.color {
+			switch l5424.SeverityLvl(severity).String() {
+			case "EmergencyLvl", "AlertLvl", "CritLvl":
+				colorOut = color.New(color.FgRed, color.Bold)
+			case "ErrorLvl":
+				colorOut = color.New(color.FgRed)
+			case "WarnLvl":
+				colorOut = color.New(color.FgYellow)
+			}
+		}
 	}
 
 	// TODO: allow for some other format delimeters?
 	// json formatting as an option?
-	return l._log(strings.Join(format, " - "), v...)
+	return l._log(colorOut, strings.Join(format, " - "), v...)
 }
 
-func (l *Logger) _log(format string, v ...interface{}) error {
-	_, err := fmt.Fprint(l.w, fmt.Sprintf("%s: %s", format, fmt.Sprint(v...)))
+func (l *Logger) _log(color *color.Color, format string, v ...interface{}) error {
+	var err error
+	if color == nil {
+		_, err = fmt.Fprint(l.w, fmt.Sprintf("%s: %s", format, fmt.Sprint(v...)))
```

Comment:
being that they have the same signature it would make sense to use an interface here so you can just set one or the other before you even get here.